doc/build/build.md

```diff
@@ -22,90 +23,103 @@ $
+* `make`
+
+On MacOS you will need to have `readlink` executable installed (`brew install coreutils`)
+
 ## 2. Building the Multi-Cluster-App-Deployer Controller
 
 ### Build the Executable
 
-Run the build script `build.sh`:
-```
-$ cd multi-cluster-app-dispatcher/deployment/
+From the root directory of the repository:
 
-$ ./build.sh
+```bash
+#build for linux OS and for use inside docker image
+multi-cluster-app-dispatcher $ make mcad-controller
 ...
-+ cd ..
-+ make generate-code
-Compiling deepcopy-gen
-Generating deepcopy
+Compiling deepcopy-gen...
+Generating deepcopy...
 go build -o _output/bin/deepcopy-gen ./cmd/deepcopy-gen/
-_output/bin/deepcopy-gen -i ./pkg/apis/controller/v1beta1/ -O zz_generated.deepcopy
-+ make kar-controller
+_output/bin/deepcopy-gen -i ./pkg/apis/controller/v1beta1/ -O zz_generated.deepcopy 
+Compiling controller
+CGO_ENABLED=0 GOOS="linux" go build -o _output/bin/mcad-controller ./cmd/kar-controllers/
+
+#build for local testing purposes, by default enable the race conditions detector
+multi-cluster-app-dispatcher $ make mcad-controller-local
+...
 mkdir -p _output/bin
-CGO_ENABLED=0 GOARCH=amd64 go build -o _output/bin/kar-controllers ./cmd/kar-controllers/
-$
+Compiling deepcopy-gen...
+Generating deepcopy...
+go build -o _output/bin/deepcopy-gen ./cmd/deepcopy-gen/
+_output/bin/deepcopy-gen -i ./pkg/apis/controller/v1beta1/ -O zz_generated.deepcopy 
+Compiling controller
+go build -race -o _output/bin/mcad-controller-local ./cmd/kar-controllers/
 ```
 
-Ensure the executables: `deepcopy-gen`, `mcad-controllers`  are created in the target output directory:
-```
-$ ls ../_output/bin/
-deepcopy-gen	mcad-controller
-$
+Ensure the executables: `deepcopy-gen`, `mcad-controllers`, and optionally `mca-controller-local` are created in the target output directory:
+
+```bash
+multi-cluster-app-dispatcher $ ls _output/bin 
+deepcopy-gen            mcad-controller         mcad-controller-local
 ```
 
 ### Build the Multi-Cluster-App-Dispatcher Image
 
-Run the image build script `image.sh`:
-
-```
-$ ./image.sh
-...
-+ make images
-Changed to executable directory
-Build the docker image
-cd ./_output/bin
-docker build --no-cache --tag mcad-controller:v1.14 ...
-Sending build context to Docker daemon  122.7MB
-Step 1/7 : From ubuntu:18.04
- ---> ea4c82dcd15a
-Step 2/7 : ADD mcad-controller /usr/local/bin
- ---> 674cefbce55a
-...
- ---> 911c7c82b5ee
-Step 7/7 : WORKDIR /usr/local/bin
- ---> Running in f2db4649e7a6
-Removing intermediate container f2db4649e7a6
- ---> 1dbf126976cf
-Successfully built 1dbf126976cf
-Successfully tagged mcad-controller:v1.14
-$
-```
+From the root directory of the repository:
 
-Note the *image name* and *image tag* from the image build script (`./image.sh`) above.  For example the *image name* and *image tag* built after running the example above is `mcad-controller:v1.14`.  List the Docker images to ensure the image exists.
+```bash
+# With docker daemon running
```

Comment:
Could some addition content be added on the diff. between `make images` and `make images-podman` to help the reader decide.  Also i do think it is helpful for a new reader executing the commands to have a way to compare with the documentation on expected results/output.  If you agree could you add some addition content for the reader to verify make command ran as expected?  Thank you!

Reply:
Done!